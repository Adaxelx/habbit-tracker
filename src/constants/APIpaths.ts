const APIurl = process.env.REACT_APP_BACKEND;

const DASHBOARD = `${APIurl}/`;
const USER = `${APIurl}/users/`;
const CALENDAR = `${DASHBOARD}calendar/`;

const APIpaths = {
  LOGIN: `${USER}login/`,
  REGISTER: `${USER}register/`,
  LOGOUT: `${USER}logout/`,
  CALENDAR,
  LABEL: `${CALENDAR}label/`,
  EVENT: `${CALENDAR}event/`,
};

export default APIpaths;
