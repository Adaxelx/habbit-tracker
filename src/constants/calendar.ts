/* eslint-disable prefer-const */

export const weekDays = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
export const weekDaysFull = [
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
  'Sunday',
];
export const months = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'Juny',
  'Juni',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const getISODate = (date: Date) => date.toISOString().match(/\d{4}-\d{2}-\d{2}/)![0];

export const getDayParsed = (date: Date) => {
  let firstDay = date.getDay();

  if (firstDay === 0) {
    firstDay = 7;
  }
  firstDay -= 1;
  return firstDay;
};

export const generateParsedDate = (date: Date) => {
  const [year, month, day] = [date.getFullYear(), date.getMonth(), date.getDate()];
  return `${year}-${month < 9 ? `0${month + 1}` : month + 1}-${day < 10 ? `0${day}` : day}`;
};

export const reversedParsedDate = (date: Date) =>
  generateParsedDate(date).split('-').reverse().join('.');

export const generateWeek = (day: Date): [Date, Date] => {
  const from = new Date(day.getTime());

  from.setDate(day.getDate() - 3);
  from.setHours(0, 0, 0, 0);

  const to = new Date(day);
  to.setDate(day.getDate() + 3);
  to.setHours(0, 0, 0, 0);
  return [from, to];
};
