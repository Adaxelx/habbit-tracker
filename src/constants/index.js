export { default as APIpaths } from './APIpaths';
export { default as paths } from './paths';
