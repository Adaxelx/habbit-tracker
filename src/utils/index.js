export { default as TestUtil } from './TestUtil';
export { default as RenderWithRouter } from './RenderWithRouter';
export * from './validators';
export * from './types.d';
