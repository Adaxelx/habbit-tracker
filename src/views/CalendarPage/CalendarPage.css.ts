import styled from 'styled-components';

export const StyledCenter = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
  @media (min-width: ${({ theme }) => theme.media.tablet.s}) {
    justify-content: flex-start;
  }
`;

export const StyledButtonWrapper = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
`;
