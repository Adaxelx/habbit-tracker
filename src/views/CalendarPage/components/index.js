export { default as CalendarTile } from './CalendarTile';
export { default as CalendarGrid } from './CalendarGrid';
export { default as CalendarNavigation } from './CalendarNavigation';
export { default as DayCard } from './DayCard';
export { default as Habbit } from './Habbit';
export { default as DayCardWrapper } from './DayCardWrapper';
export { default as HabbitForm } from './HabbitForm';
export { default as WeekDaysInput } from './WeekDaysInput';
export { default as LabelForm } from './LabelForm';
export { default as LabelList } from './LabelList';
export { default as CalendarGridView } from './CalendarGridView';
