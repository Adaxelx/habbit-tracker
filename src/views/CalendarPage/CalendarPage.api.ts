import APIpaths from 'constants/APIpaths';
import { EventSend, LabelSend } from 'utils';

export const getEvents = async (token: string | undefined, from: string, to: string) => {
  const response = await fetch(`${APIpaths.EVENTS}?from=${from}&to=${to}`, {
    headers: { 'Content-Type': 'application/json', Authorization: `${token}` },
    method: 'GET',
  });

  if (response.status === 200) {
    return response.json();
  }

  const { message } = await response.json();
  throw new Error(message);
};

export const getLabels = async (token: string | undefined) => {
  const response = await fetch(`${APIpaths.LABELS}`, {
    headers: { 'Content-Type': 'application/json', Authorization: `${token}` },
    method: 'GET',
  });

  if (response.status === 200) {
    return response.json();
  }

  const { message } = await response.json();
  throw new Error(message);
};

export const postEvent = async (token: string | undefined, event: EventSend) => {
  const response = await fetch(`${APIpaths.EVENTS}`, {
    headers: { 'Content-Type': 'application/json', Authorization: `${token}` },
    method: 'POST',
    body: JSON.stringify(event),
  });

  if (response.status === 201) {
    return true;
  }

  const { message } = await response.json();
  throw new Error(message);
};

export const postLabel = async (token: string | undefined, label: LabelSend) => {
  const response = await fetch(`${APIpaths.LABELS}`, {
    headers: { 'Content-Type': 'application/json', Authorization: `${token}` },
    method: 'POST',
    body: JSON.stringify(label),
  });

  if (response.status === 201) {
    return true;
  }

  const { message } = await response.json();
  throw new Error(message);
};
