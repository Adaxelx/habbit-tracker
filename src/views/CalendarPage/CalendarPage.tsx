import React, { useEffect, useState } from 'react';
import { useUserContext } from 'context';
import { getISODate, Event, Label } from 'constants/calendar';
import { StyledCenter } from './CalendarPage.css';
import { CalendarGrid } from './components';
import { getEvents, getLabels } from './CalendarPage.api';

/* eslint-disable */

const CalendarPage = () => {
  const [date, setDate] = useState(new Date());
  const [actualMonth, setActualMonth] = useState(date.getMonth());
  const [actualYear, setActualYear] = useState(date.getFullYear());
  const [events, setEvents] = useState([]);
  const [labels, setLabels] = useState([]);
  const { token } = useUserContext();

  useEffect(() => {
    const fetchEvents = async () => {
      try {
        const from = getISODate(new Date(actualYear, actualMonth, 2));
        const daysInMonth = 33 - new Date(actualYear, actualMonth, 32).getDate();
        const to = getISODate(new Date(actualYear, actualMonth, daysInMonth));
        const response = await getEvents(token, from, to);
        setEvents(response);
      } catch (err) {
        console.log(err);
      }
    };
    fetchEvents();
  }, []);

  useEffect(() => {
    const fetchEvents = async () => {
      try {
        const response = await getLabels(token);
        setLabels(response);
        const mappedResponse = response.map((event: Event) => {
          const label = response.filter((label: Label) => event.label === label._id)[0];
          return { ...event, label };
        });
      } catch (err) {
        console.log(err);
      }
    };
    fetchEvents();
  }, []);
  console.log(events);
  return (
    <StyledCenter>
      <CalendarGrid events={events} />
    </StyledCenter>
  );
};

export default CalendarPage;
