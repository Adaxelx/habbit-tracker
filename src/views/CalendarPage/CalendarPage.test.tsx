import React from 'react';
import 'jest-styled-components';
import { waitFor, act } from '@testing-library/react';
import APIpaths from 'constants/APIpaths';
import { TestUtil, Event, Label } from 'utils';
import { months } from 'constants/calendar';
import CalendarPage from './CalendarPage';

const failMessage = (message: string) =>
  Promise.resolve({
    status: 500,
    ok: false,
    json: () => Promise.resolve({ message }),
  });

describe('CalendarPage', () => {
  let util: TestUtil;
  let failE: boolean;
  let failL: boolean;
  let events: Event[];
  let labels: Label[];

  const mockedFetch = (input: string, init: { method: string }) => {
    if (failE && input.includes(APIpaths.EVENTS)) {
      return failMessage('failEvent');
    }
    if (failL && input.includes(APIpaths.LABELS)) {
      return failMessage('failLabel');
    }

    switch (init.method) {
      case 'GET':
        if (input.includes(APIpaths.EVENTS)) {
          return Promise.resolve({
            status: 200,
            ok: true,
            json: () => Promise.resolve(events),
          });
        }
        return Promise.resolve({
          status: 200,
          ok: true,
          json: () => Promise.resolve(labels),
        });

      default:
        return null;
    }
  };

  beforeAll(() => {
    jest.spyOn(window, 'fetch');
  });
  beforeEach(() => {
    jest.clearAllMocks();
    window.fetch.mockImplementation(mockedFetch);

    failE = false;
    failL = false;
    labels = [
      {
        userId: 'abd',
        color: '#fff',
        _id: 'xd',
        title: 'siema',
      },
    ];
    events = [
      {
        checked: [{ day: 3, month: 3, year: 2021, _id: 'xdd' }],
        dateEnd: '2021-04-30',
        dateStart: '2021-04-03',
        daysOfWeek: [0, 1, 3],
        description: 'opis',
        timeStart: '12:12',
        timeEnd: '12:30',
        label: {
          color: '#fff',
          _id: 'xdd',
          title: 'siema',
          userId: 'abd',
        },
        _id: '123',
        title: 'xdd',
        userId: '123',
      },
    ];
  });

  it('should match snapshot (mobile)', async () => {
    global.innerWidth = 375;
    const mockDate = new Date('2022-01-01');
    const spy = jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
    act(() => {
      util = new TestUtil(<CalendarPage />);
    });

    await waitFor(() => expect(fetch).toHaveBeenCalledTimes(2));
    expect(util.render.asFragment()).toMatchSnapshot();
    spy.mockRestore();
  });

  it('should show error message event', async () => {
    failE = true;
    act(() => {
      util = new TestUtil(<CalendarPage />);
    });

    await waitFor(() => expect(fetch).toHaveBeenCalledTimes(2));
    expect(util.getAll('alert')[0].textContent).toBe('failEvent');
  });

  it('should move dates', async () => {
    act(() => {
      util = new TestUtil(<CalendarPage />);
    });

    await waitFor(() => expect(fetch).toHaveBeenCalledTimes(2));
    const [month, year] = util.get('dateCalendar-main')?.textContent?.split(' ') as [
      string,
      string,
    ];
    const index = months.findIndex((monthArr) => monthArr === month);
    const prevIndex = index - 1 < 0 ? 11 : index - 1;
    const prevYear = index - 1 < 0 ? year - 1 : year;

    util.click('moveLeft-main');
    expect(util.get('dateCalendar-main').textContent).toBe(`${months[prevIndex]} ${prevYear}`);

    util.click('moveRight-main');
    expect(util.get('dateCalendar-main').textContent).toBe(`${month} ${year}`);
  });
});
