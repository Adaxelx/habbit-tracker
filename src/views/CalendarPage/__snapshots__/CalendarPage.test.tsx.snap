// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarPage should match snapshot (mobile) 1`] = `
<DocumentFragment>
  .c9 {
  color: #fff;
  background-color: #FFC136;
  border: 1px solid #707070;
  padding: 0.75rem;
  font-size: 1rem;
  min-width: none;
  display: inline-block;
  text-align: center;
  margin-top: 16px;
  margin-left: 0;
  margin-right: 0;
}

.c9:hover {
  cursor: pointer;
}

.c9:disabled {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c10 {
  color: #fff;
  background-color: #FFC136;
  border: 1px solid #707070;
  padding: 0.75rem;
  font-size: 1rem;
  min-width: none;
  display: inline-block;
  text-align: center;
  margin-top: 16px;
  margin-left: 16px;
  margin-right: 16px;
}

.c10:hover {
  cursor: pointer;
}

.c10:disabled {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c15 {
  color: #fff;
  background-color: #FFC136;
  border: 1px solid #707070;
  padding: 0.75rem;
  font-size: 1rem;
  min-width: none;
  display: inline-block;
  text-align: center;
  margin-top: 0;
  margin-left: 0.75rem;
  margin-right: 0.75rem;
}

.c15:hover {
  cursor: pointer;
}

.c15:disabled {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c16 {
  color: #000000;
  background-color: #FF9191;
  border: 1px solid #707070;
  padding: 0.75rem;
  font-size: 1rem;
  min-width: none;
  display: inline-block;
  text-align: center;
  margin-top: 0;
  margin-left: 0;
  margin-right: 0;
}

.c16:hover {
  cursor: pointer;
}

.c16:disabled {
  -webkit-filter: grayscale(1);
  filter: grayscale(1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 100vh;
  max-width: 100%;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  margin-top: 5rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(7,minmax(0,1fr));
  grid-template-rows: auto;
  max-width: 100%;
  border: 1px solid #707070;
  border-top: none;
  background-color: #fff;
}

.c6 {
  background-color: #EDEFFF;
  border-top: 1px solid #707070;
  border-bottom: none;
  justify-items: center;
}

.c7 {
  width: 50px;
  padding: 0.75rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0.75rem 0;
}

.c3 {
  background-color: transparent;
  border: none;
  font-size: 1.5rem;
}

.c4 {
  width: 200px;
  text-align: center;
  line-height: 100%;
  margin: 0;
}

.c12 {
  background-color: #AFE9FF;
  min-width: 275px;
  min-height: 500px;
  border: 1px solid #707070;
}

.c13 {
  font-size: 1.5rem;
  margin: 1.25rem 0;
  text-align: center;
}

.c14 {
  margin-bottom: 0.5rem;
}

.c18 {
  background-color: #fff;
  padding: 0.75rem;
  width: auto;
  border-radius: 20px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0 0.75rem;
  border-left: 1px solid #77CBFF;
  margin: 0 0.75rem;
  margin-top: 0.5rem;
}

.c17:hover {
  cursor: pointer;
}

.c19 {
  font-size: 2rem;
  margin-bottom: 0.5rem;
  margin-top: 0;
}

.c20 {
  color: gray;
  margin-bottom: 0.5rem;
}

.c21 {
  text-align: justify;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 100%;
  overflow-x: scroll;
}

@media (min-width:768px) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding-top: 0;
  }
}

@media (min-width:768px) {
  .c4 {
    width: 400px;
    font-size: 2rem;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <button
          class="c3"
          data-testid="moveLeft-main"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-left fa-w-6 "
            data-icon="caret-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
              fill="currentColor"
            />
          </svg>
        </button>
        <h3
          class="c4"
          data-testid="dateCalendar-main"
        >
          March 2022
        </h3>
        <button
          class="c3"
          data-testid="moveRight-main"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-caret-right fa-w-6 "
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="c5 c6"
      >
        <div
          class="c7"
        >
          Mon
        </div>
        <div
          class="c7"
        >
          Tue
        </div>
        <div
          class="c7"
        >
          Wed
        </div>
        <div
          class="c7"
        >
          Thu
        </div>
        <div
          class="c7"
        >
          Fri
        </div>
        <div
          class="c7"
        >
          Sat
        </div>
        <div
          class="c7"
        >
          Sun
        </div>
      </div>
      <div
        class="c5"
      />
    </div>
    <div
      class="c8"
    >
      <button
        class="c9"
        data-testid="addh"
      >
        Add habbit
      </button>
      <button
        class="c10"
        data-testid="addl"
      >
        Add label
      </button>
      <button
        class="c9"
        data-testid="labell"
      >
        Label list
      </button>
    </div>
    <h2>
      25.03.2022 - 25.03.2022
    </h2>
    <div
      class="c11"
    >
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
      </div>
      <div
        class="c12"
      >
        <h3
          class="c13"
        >
          Friday
        </h3>
        <div
          class="c14"
        >
          <button
            class="c15"
            data-testid="edit"
          >
            Edit habbit
          </button>
          <button
            class="c16"
            data-testid="delete"
          >
            X
          </button>
          <div
            class="c17"
            data-testid="habbit123"
          >
            <small
              class="c18"
              color="#fff"
            >
              siema
            </small>
            <h2
              class="c19"
            >
              xdd
            </h2>
            <p
              class="c20"
            >
              12:12-12:30
            </p>
            <p
              class="c21"
            >
              opis
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
